### DISCUSSION TITLE
How to run migrations without transaction context?

### DISCUSSION THREAD

**Discussion by frol** [diesel-rs/diesel#2841] [Q&A] [answered]
Adding new indices in Postgres you may choose to use `CREATE INDEX CONCURRENTLY` to avoid locking the table, however, if you use it in Diesel migration, it fails with:

```
CREATE INDEX CONCURRENTLY cannot run inside a transaction block
```

Can I opt out of running some migrations inside a transaction block?

**Comment by weiznich** [comment, marked as answer]
That's not supported with diesel 1.4.x. The master branch supports something like this with the rewrite of the migration crate in https://github.com/diesel-rs/diesel/pull/2657

**Comment by tversteeg** [comment]
This is currently supported with file-based migrations by adding a `metadata.toml` file in your migration directory with the following line:

```toml
run_in_transaction = false
```
